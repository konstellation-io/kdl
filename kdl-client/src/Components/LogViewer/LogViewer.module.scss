@import 'Styles/colors';
@import 'Styles/variables';
@import 'Styles/mixins';

.wrapper {
  @include font-code;

  overflow-x: hidden;
  overflow-y: scroll;
}

.wrapper p {
  margin-right: 2.5 * $grid-unit;
  white-space: pre-wrap; // Draw as <BR> the new lines in react string
}

.log {
  color: font-color(lowlight);
  padding-left: 2.5 * $grid-unit;

  &.error {
    @include circle-before(palette(error));

    color: palette(error);
    padding-left: 2.5 * $grid-unit;
  }
}
