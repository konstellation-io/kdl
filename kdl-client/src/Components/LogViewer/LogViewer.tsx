import React from 'react';
import cx from 'classnames';
import styles from './LogViewer.module.scss';

export type Log = {
  text: string;
  isError?: boolean;
};

interface Props {
  logs: Log[];
}

function LogViewer({ logs }: Props) {
  return (
    <div className={styles.wrapper}>
      {logs.map(({ text, isError = false }, idx) => (
        <p key={idx} className={cx(styles.log, { [styles.error]: isError })}>
          {text}
        </p>
      ))}
    </div>
  );
}

export default LogViewer;
