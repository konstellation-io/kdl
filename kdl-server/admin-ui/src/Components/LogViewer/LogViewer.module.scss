@import 'Styles/colors';
@import 'Styles/variables';
@import 'Styles/mixins';

.wrapper {
  @include font-code;

  overflow: scroll;
  overflow-x: hidden;
  height: 100%;
}

.wrapper p {
  margin-right: 2.5 * $grid-unit;
  white-space: pre-wrap; // Draw as <BR> the new lines in react string
}

.log {
  color: font-color(lowlight);
  padding-left: 2.5 * $grid-unit;

  &.error {
    @include circle-before(palette(error));

    color: palette(error);
    padding-left: 2.5 * $grid-unit;
  }
}
