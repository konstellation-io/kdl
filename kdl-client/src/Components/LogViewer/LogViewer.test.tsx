import LogViewer from './LogViewer';
import React from 'react';
import { shallow } from 'enzyme';

const logs = [
  {
    text: 'log line 1',
    isError: false,
  },
  {
    text: 'log line 2',
    isError: true,
  },
];

const component = shallow(<LogViewer logs={logs} />);

describe('LogViewer component', () => {
  test('Component match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('show right texts', () => {
    expect(component.contains('log line 1')).toBeTruthy();
    expect(component.contains('log line 2')).toBeTruthy();
  });

  it('show error logs', () => {
    expect(component.find('.error').length).toBe(1);
  });
});
