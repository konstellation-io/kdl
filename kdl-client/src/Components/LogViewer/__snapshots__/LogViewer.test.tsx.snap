// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogViewer component Component match snapshot 1`] = `
<div
  className="wrapper"
>
  <p
    className="log"
    key="0"
  >
    log line 1
  </p>
  <p
    className="log error"
    key="1"
  >
    log line 2
  </p>
</div>
`;
